cyberdrop_dl/scraper/crawlers/nekohouse_crawler.py

```diff
@@ -0,0 +1,270 @@
+                        "name": file['href'].split("?f=")[-1] if "?f=" in file['href'] else file['href'].split("/")[-1].split("?")[0]
+                    }
+                    data["attachments"].append(attachment)
+
+                for file in soup.select(self.post_videos_selector):
+                    attachment = {
+                        "path": file['src'].replace('/data/', 'data/'),
+                        "name": file['src'].split("?f=")[-1] if "?f=" in file['src'] else file['src'].split("/")[-1].split("?")[0]
+                    }
+                    data["attachments"].append(attachment)
+
+                for file in soup.select(self.file_downloads_selector):
+                    attachment = {
+                        "path": file['href'].replace('/data/', 'data/'),
+                        "name": file['href'].split("?f=")[-1] if "?f=" in file['href'] else file['href'].split("/")[-1].split("?")[0]
+                    }
+                    data["file"].append(attachment)
+        else:
+            async with self.request_limiter:
+                soup: BeautifulSoup = await self.client.get_BS4(self.domain, post_url)
+                # Published as current time to avoid errors.
+                data = {
+                    "id": post,
+                    "user": user,
+                    "service": service,
+                    "title": soup.select_one('title').text.strip(),
+                    "content": "Unknown",
+                    "user_str": user_str,
+                    "published": datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
+                    "file": [],
+                    "attachments": []
+                }
+                
+                for file in soup.select("a[class=post__attachment-link]"):
+                    attachment = {
+                        "path": file['href'].replace('/data/', 'data/'),
+                        "name": file['href'].split("?f=")[-1] if "?f=" in file['href'] else file['href'].split("/")[-1].split("?")[0]
+                    }
+                    data["attachments"].append(attachment)
+
+        await self.handle_post_content(scrape_item, data, user, user_str)
+
+
+    @error_handling_wrapper
+    async def handle_post_content(self, scrape_item: ScrapeItem, post: Dict, user: str, user_str: str) -> None:
+        """Handles the content of a post"""
+        date = post["published"].replace("T", " ")
+        post_id = post["id"]
+        post_title = post.get("title", "")
+
+        scrape_item.album_id = post_id 
+        scrape_item.part_of_album = True
+
+        async def handle_file(file_obj):
+            link = self.primary_base_domain / file_obj['path']
+            link = link.with_query({"f": file_obj['name']})
+            await self.create_new_scrape_item(link, scrape_item, user_str, post_title, post_id, date)
+
+        for file in post['attachments']:
+            await handle_file(file)
```

Comment:
These are final file URLs. Better to just send them straight into `self.handle_direct_link()`, not back to the scrape queue